src/config.cpp

```diff
@@ -546,11 +546,55 @@ class SimulationConfig::Parser
             auto& valueIt = it.value();
             const auto debugStr = fmt::format("report {}", it.key());
             parseMandatory(valueIt, "cells", debugStr, report.cells);
+            parseOptional(valueIt, "sections", report.sections);
             parseMandatory(valueIt, "type", debugStr, report.type);
+            parseOptional(valueIt, "scaling", report.scaling);
+            parseOptional(valueIt, "compartments", report.compartments);
+            parseMandatory(valueIt, "variable_name", debugStr, report.variableName);
+            parseOptional(valueIt, "unit", report.unit);
             parseMandatory(valueIt, "dt", debugStr, report.dt);
             parseMandatory(valueIt, "start_time", debugStr, report.startTime);
             parseMandatory(valueIt, "end_time", debugStr, report.endTime);
             parseOptional(valueIt, "file_name", report.fileName);
+            parseOptional(valueIt, "enabled", report.enabled);
+
+            if (report.sections.empty()) {
+                report.sections = "soma";
+            } else {
+                std::set<std::string> possibleValues = {"soma", "axon", "dend", "apic", "all"};
```

Comment:
can we add a helper function to do this test?  I think it would be nice to be able to do it with having to create a `std::set`, too.

Reply:
ahem, that's *without* creating a std::set.

Reply:
an `initializer_list` probably works:
```
template<typename T>
bool isin(const T& needle, std::initializer_list<T>haystack)
{
    return std::find(std::begin(haystack),
                     std::end(haystack),
                     needle) != std::end(haystack);
}
```

Reply:
I tried but I was getting errors.
Apparently it doesn't work with strings:
https://www.geeksforgeeks.org/efficient-ways-to-compare-a-variable-with-multiple-values/

Its a pity we don't have C++17 since there is a more elegant solution...
```
template <typename First, typename... T>
bool is_in(First&& first, T&&... t)
{
    return ((first == t) || ...);
}
string c = "abc";
// Compare using template
if (is_in(c, "xyz", "bhy", "abc"))
```

Reply:
actually with initializer_list it would work but you need to cast it
```
std::string value = "val1";
if (is_in(val, std::initializer_list<std::string>{ "val1", "val2", "val3" })) {
...
```

Reply:
doesn't it work by giving the template param:

ie: `if (is_in(c, "xyz", "bhy", "abc"))` ->`if (is_in<std::string>(c, "xyz", "bhy", "abc"))`